release-19.2: backupccl,importccl: fix privilege checks for BACKUP/RESTORE and IMPORT

pbardea opened the pull request:
Backport 2/2 commits from #44250.

/cc @cockroachdb/release

---

This changes the privilege checks in IMPORT, IMPORT INTO and RESTORE to
run during the *planning* of the job, in the SQL plan hook execution,
rather than during the execution of the job. This is done because
privilege checks are implemented on planner, and close over the
planner's txn in some branches/cases, so invoking them later, on a
txn-less planner in a resumed jobs execution, can cause problems.

Before this, the planStateHook's txn was assumed to be set and caused a
panic on checking RBAC privileges. Additionally, permission checks in these
operations did not properly give access to all admin users.

Fixes #44252.

Release note (bug fix): Allow all admin users to use BACKUP/RESTORE and
IMPORT.

pbardea committed:
import,restore: add RBAC tests with non-root admin users

Add failing tests for a non-root admin user trying to IMPORT{,INTO},
BACKUP and RESTORE. These types of users should be allowed to do these
operations but we have found issues with permissions not letting them as
well as panics due to incorrect usage of the planner in IMPORT INTO.

Release note: None

pbardea committed:
import,restore: move all privilege checks to run during job planning

This changes the privilege checks in IMPORT, IMPORT INTO and RESTORE to
run during the *planning* of the job, in the SQL plan hook execution,
rather than during the execution of the job. This is done because
privilege checks are implemented on planner, and close over the
planner's txn in some branches/cases, so invoking them later, on a
txn-less planner in a resumed jobs execution, can cause problems.

Release note (bug fix): Allow all admin users to use BACKUP/RESTORE and
IMPORT.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/44456)
<!-- Reviewable:end -->

pbardea committed:
backupccl: include role package in backup test

Include the role package in the backup test so that the tests have
access to the appropriate binary allowing to test BACKUP/RESTORE
interaction with RBAC.

Release note: None

pbardea commented:
bors r+
